extensions-core/kafka-indexing-service/src/main/java/io/druid/indexing/kafka/supervisor/KafkaSupervisor.java

```diff
@@ -335,6 +359,8 @@ public void stop(boolean stopGracefully)
         }
         log.info("Shutdown notice handled");
 
+        taskClient.close();
+        workerExec.shutdown();
```

Comment:
Hm, I was probably thinking that using shutdown here (and in the task client) would give more chance for any tasks that needed to be stopped/killed to be handled correctly if they didn't respond before the shutdown timeout, but I'm not sure it makes much difference and it's probably more important to ensure the overlord doesn't have any stuck threads hanging around so I agree shutdownNow is better here.